### DISCUSSION TITLE
Named entitiy recognition using spans vs using doc tokens.

### DISCUSSION THREAD

**Discussion by paulie8** [explosion/spaCy#11346] [Help: Other Questions]
I have been using a document of Peter Rabbit to finish some assignments. So I was given the question of finding the number of sentences that contained named entities. So I went ahead and created a list of sentence span tokens.

lista=[sent for sent in doc.sents]

After this, I used this list to create a list of sentences having NERs using:

listents=[i for i in lista if (i.ents)]

I checked the length of listents and found it to be 53. All well uptil now. 
Now what I tried doing after this was chaning lista into a list of spacy doc objects instead of span tokens using a slight change in the code:
lista=[nlp(sent.text) for sent in doc.sents]
 ##nlp here is my object connecting to the english language library.

Now here is the interesting thing. After running this line, my listents obviously got updated and repeated finding its length. To my surprise I got a length of 55 (instead of the previous 53 whilst using span tokens).
I tried printing the entities from both the methods and found that while using sentence span tokens and finding the named entities in the list, spacy was ignoring 2 entities.
I know that that for a list comprising of span tokens if we find token.ents we get tuples and if we had used a list of doc objects, we get a list. However, I dont seem to understand why spacy is excluding 2 named entities when I am using span tokens.

<img width="455" alt="sth" src="https://user-images.githubusercontent.com/105962077/185582689-f3b85363-95f4-491f-ad7a-6b9c0ec27466.png">

The text in bold on the left doesnt appear when I call .ents from the span list.

Please correct me if I am wrong.

**Comment by rmitsch** [comment]
Hi @paulie8, 

to understand the behavior you are describing, keep in mind that spaCy's NER component is a trained statistical model taking into account the context of individual tokens. 
If you split your text into sentences and analyze them individually, it might very well happen that the model's decisions on what constitutes an entity diverge. This is due to the model not having access to the same context it has when processing the entire document at once.

Since this is not an issue, I'll convert this to a discussion.

**Reply by paulie8** [reply]
Okay, that's very intersting.
Just one more thing. Are there any set rules which determine how spacy distinguishes between sentences and doc objects whilst finding context for ner (or rather any component of the pipeline)?
Thank you!

**Reply by rmitsch** [reply]
The concept of _distinction_ between sentences and doc objects doesn't apply here, since the former are a part of the latter. The NER component looks at one doc object at a time. You segmented your original document into many doc objects, each representing one sentence.
